FreeRTOS/FreeRTOS-Kernel
# Warning fixes.

Warning fixes in timers.c and tasks.c

Description
-----------
Fixes the following warnings:
```
arm-none-eabi-gcc -mcpu=cortex-m4 -mthumb -mfloat-abi=hard -mfpu=fpv4-sp-d16 -O2 -fmessage-length=0 -fsigned-char -ffunction-sections -fdata-sections -Wunused -Wuninitialized -Wall -Wextra -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Waggregate-return -Wfloat-equal  -g -D_RENESAS_RA_ -I"/e2_studio/workspace/freertos/src" -I"/e2_studio/workspace/freertos/ra/fsp/inc" -I"/e2_studio/workspace/freertos/ra/fsp/inc/api" -I"/e2_studio/workspace/freertos/ra/fsp/inc/instances" -I"/e2_studio/workspace/freertos/ra/fsp/src/rm_freertos_port" -I"/e2_studio/workspace/freertos/ra/aws/amazon-freertos/freertos_kernel/include" -I"/e2_studio/workspace/freertos/ra/arm/CMSIS_5/CMSIS/Core/Include" -I"/e2_studio/workspace/freertos/ra_gen" -I"/e2_studio/workspace/freertos/ra_cfg/fsp_cfg/bsp" -I"/e2_studio/workspace/freertos/ra_cfg/fsp_cfg" -I"/e2_studio/workspace/freertos/ra_cfg/aws" -std=c99 -MMD -MP -MF"ra/aws/amazon-freertos/freertos_kernel/timers.d" -MT"ra/aws/amazon-freertos/freertos_kernel/timers.o" -c -o "ra/aws/amazon-freertos/freertos_kernel/timers.o" -x c "../ra/aws/amazon-freertos/freertos_kernel/timers.c"
../ra/aws/amazon-freertos/freertos_kernel/timers.c: In function 'vTimerSetReloadMode':
../ra/aws/amazon-freertos/freertos_kernel/timers.c:456:38: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  456 |                 pxTimer->ucStatus &= ~tmrSTATUS_IS_AUTORELOAD;
      |                                      ^
../ra/aws/amazon-freertos/freertos_kernel/timers.c: In function 'prvProcessExpiredTimer':
../ra/aws/amazon-freertos/freertos_kernel/timers.c:542:34: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  542 |             pxTimer->ucStatus &= ~tmrSTATUS_IS_ACTIVE;
      |                                  ^
../ra/aws/amazon-freertos/freertos_kernel/timers.c: In function 'prvProcessReceivedCommands':
../ra/aws/amazon-freertos/freertos_kernel/timers.c:840:46: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  840 |                         pxTimer->ucStatus &= ~tmrSTATUS_IS_ACTIVE;
      |                                              ^
../ra/aws/amazon-freertos/freertos_kernel/timers.c:879:54: warning: conversion from 'int' to 'uint8_t' {aka 'unsigned char'} may change value [-Wconversion]
  879 |                                 pxTimer->ucStatus &= ~tmrSTATUS_IS_ACTIVE;
      |                                                      ^
arm-none-eabi-gcc -mcpu=cortex-m4 -mthumb -mfloat-abi=hard -mfpu=fpv4-sp-d16 -O2 -fmessage-length=0 -fsigned-char -ffunction-sections -fdata-sections -Wunused -Wuninitialized -Wall -Wextra -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Waggregate-return -Wfloat-equal  -g -D_RENESAS_RA_ -I"/e2_studio/workspace/freertos/src" -I"/e2_studio/workspace/freertos/ra/fsp/inc" -I"/e2_studio/workspace/freertos/ra/fsp/inc/api" -I"/e2_studio/workspace/freertos/ra/fsp/inc/instances" -I"/e2_studio/workspace/freertos/ra/fsp/src/rm_freertos_port" -I"/e2_studio/workspace/freertos/ra/aws/amazon-freertos/freertos_kernel/include" -I"/e2_studio/workspace/freertos/ra/arm/CMSIS_5/CMSIS/Core/Include" -I"/e2_studio/workspace/freertos/ra_gen" -I"/e2_studio/workspace/freertos/ra_cfg/fsp_cfg/bsp" -I"/e2_studio/workspace/freertos/ra_cfg/fsp_cfg" -I"/e2_studio/workspace/freertos/ra_cfg/aws" -std=c99 -MMD -MP -MF"ra/aws/amazon-freertos/freertos_kernel/tasks.d" -MT"ra/aws/amazon-freertos/freertos_kernel/tasks.o" -c -o "ra/aws/amazon-freertos/freertos_kernel/tasks.o" -x c "../ra/aws/amazon-freertos/freertos_kernel/tasks.c"
../ra/aws/amazon-freertos/freertos_kernel/tasks.c: In function 'prvDeleteTCB':
../ra/aws/amazon-freertos/freertos_kernel/tasks.c:3932:39: warning: unused parameter 'pxTCB' [-Wunused-parameter]
 3932 |     static void prvDeleteTCB( TCB_t * pxTCB )
      |                               ~~~~~~~~^~~~~
```

Test Steps
-----------
Build using GCC (gcc-arm-none-eabi-9-2020-q2-update) with all warnings enabled.

Related Issue
-----------
See above. I have not made a GitHub issue, but I will if you request it.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
